src/value/borrowed.rs

```diff
@@ -293,24 +294,29 @@ impl<'val> IonStruct for StructRef<'val> {
         )
     }
 
-    fn get<T: AsRef<str>>(&self, field_name: T) -> Option<&Self::Element> {
-        self.text_fields
-            .get(field_name.as_ref())?
-            .last()
-            .map(|(_s, v)| v)
+    fn get<T: AsSymbolRef>(&self, field_name: T) -> Option<&Self::Element> {
+        if let Some(text) = field_name.as_symbol_ref().text() {
+            self.text_fields.get(text)?.last().map(|(_s, v)| v)
+        } else {
+            self.no_text_fields.last().map(|(_name, value)| value)
```

Comment:
Should we remove `no_text_fields` from here as well?

Reply:
I intentionally avoided overhauling `StructRef` in this PR (but then forgot to make that explicit! 😬) I plan to make the same change in a follow-on if we're ok with this.

Reply:
That works. We can do it in a separate PR.